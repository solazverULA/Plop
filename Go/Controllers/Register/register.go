package register

import (
	//"bytes"
	"encoding/json"
	//"fmt"
	"log"
	"net/http"

	"../../Controllers/User"
	"../../Models/ModelUser"
)

//Funcion para registrar un nuevo usuario
func Register(w http.ResponseWriter, r *http.Request) {

	user, people, cities, countries, filelogo, handlelogo, _, _ := user.GetDataFromUser(r)
	//user, people, cities, countries := user.GetUserRequest(r)

	w.Header().Set("Content-Type", "text/html; charset-utf-8")
	w.Header().Set("Access-Control-Allow-Origin", "*")

	var responseUser modeluser.ResponseUser

	val := modeluser.GetUsuario(user.Email)

	if !modeluser.ValideEmail(val.Email, user) {
			user_creado := modeluser.CreateUser(user, people, cities, countries, filelogo, handlelogo)
			//user_creado := modeluser.CreateUser(user, people, cities, countries, nil, nil)
			responseUser = modeluser.ResponseUser{"success", user_creado, "¡Usuario registrado con éxito!"}
			log.Println("Registro")
			json.NewEncoder(w).Encode(responseUser)
	} else {
		responseUser = modeluser.ResponseUser{"error", user, "Usuario ya existe."}
		json.NewEncoder(w).Encode(responseUser)
	}

}


//fluency.team@openenglish.com
//11.200 Bs 
//link azul. 