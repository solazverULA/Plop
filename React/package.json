{
  "name": "nud-react",
  "version": "1.0.0",
  "private": true,
  "homepage": "https://timcreative.github.io/now-ui-dashboard-react",
  "dependencies": {
    "animate.css": "^3.6.1",
    "babel-preset-env": "^1.7.0",
    "babel-preset-react": "^6.24.1",
    "bootstrap": "4.0.0-beta",
    "chart.js": "^2.7.2",
    "history": "4.7.2",
    "jquery": "3.3.1",
    "node-sass": "^4.9.0",
    "node-sass-chokidar": "^1.3.0",
    "npm-run-all": "4.1.1",
    "perfect-scrollbar": "1.3.0",
    "react": "16.1.0",
    "react-animated-css": "^1.0.4",
    "react-chartjs-2": "2.7.0",
    "react-dom": "16.1.0",
    "react-google-maps": "9.2.2",
    "react-notification-alert": "0.0.7",
    "react-router-dom": "4.2.2",
    "react-scripts": "^1.1.4",
    "reactstrap": "5.0.0-alpha.4"
  },
  "devDependencies": {
    "babel-jest": "^22.0.4",
    "jest": "^22.0.4",
    "react-test-renderer": "^16.0.0"
  },
  "scripts": {
    "build-css": "node-sass-chokidar --include-path ./src --include-path ./node_modules src/ -o src/",
    "watch-css": "npm run build-css && node-sass-chokidar --include-path ./src --include-path ./node_modules src/ -o src/ --watch --recursive",
    "start-js": "react-scripts start",
    "start": "npm-run-all -p watch-css start-js",
    "build": "npm run build-css && react-scripts build",
    "test": "jest",
    "jest-update-snapshot": "jest --updateSnapshot",
    "eject": "react-scripts eject"
  }
}
