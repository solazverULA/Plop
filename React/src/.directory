[Dolphin]
Timestamp=2018,6,27,3,57,52
Version=3
