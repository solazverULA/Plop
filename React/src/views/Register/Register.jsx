import React from 'react';
import {
    Card, CardHeader, Button, CardBody, CardFooter, CardTitle, Row, Col, UncontrolledDropdown, DropdownToggle, DropdownMenu, DropdownItem, Table, Input,  InputGroup, FormGroup
} from 'reactstrap';
// react plugin used to create charts
import { Line, Bar } from 'react-chartjs-2';
// function that returns a color based on an interval of numbers

import {
    PanelHeader, Stats, CardCategory, Tasks
} from '../../components';

import {
    dashboardPanelChart,
    dashboardShippedProductsChart,
    dashboardAllProductsChart,
    dashboard24HoursPerformanceChart
} from '../../variables/charts.jsx';

import { tasks } from '../../variables/general.jsx';

class Register extends React.Component{
    render(){
        return (
            <div>
                <PanelHeader
                    size="sm"
                />
              <div className="content">
                    <Row className="justify-content-center">
                        <Col xs={12} md={4} className="justify-content-center">
                            <Card className="card-chart" style={{height: '300px'}}>
                                <CardHeader style={{paddingTop:'7%'}}>
                                    <CardTitle>Register</CardTitle>
                                </CardHeader>
                                <CardBody>
                                    <div xs={10} md={2} lg={2} style={{paddingTop:'2%'}}>
                                      <FormGroup>
                                        <div>
                                          E-mail
                                        </div>
                                        <InputGroup>
                                          <Input type="email" id="register_email" name="Email" required={true}/>
                                        </InputGroup>
                                      </FormGroup>
                                      <FormGroup>
                                        <div>
                                          Password
                                        </div>
                                        <InputGroup>
                                          <Input type="password" id="register_password" name="Password" required={true}/>
                                        </InputGroup>
                                      </FormGroup>
                                      <FormGroup>
                                        <div>
                                          Confirm password
                                        </div>
                                        <InputGroup>
                                          <Input type="password" id="register_confirm_password" name="Password" required={true}/>
                                        </InputGroup>
                                      </FormGroup>
                                      <FormGroup>
                                        <div>
                                          ID number
                                        </div>
                                        <InputGroup>
                                          <Input type="number" id="register_id_number" name="id" required={true}/>
                                        </InputGroup>
                                      </FormGroup>
                                      <FormGroup>
                                        <div>
                                          Gender
                                        </div>
                                        <InputGroup>
                                          <Input type="number" id="register_id_number" name="id" required={true}/>
                                        </InputGroup>
                                      </FormGroup>
                                    </div>
                                </CardBody>
                                <CardFooter className='justify-content-right text-right' style={{marginTop:'-2%'}}>
                                  <Button color="primaryBlue" style={{width:'100px'}}>register</Button>
                                </CardFooter>
                            </Card>
                        </Col>
                    </Row>
                </div>
            </div>
        );
    }
}

export default Register;
