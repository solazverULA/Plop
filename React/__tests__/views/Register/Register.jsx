import React from 'react';
import Register from '../../../src/views/Register/Register';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const loading = renderer.create(
    <Register/>
  ).toJSON();
  
  expect(loading).toMatchSnapshot();
});
