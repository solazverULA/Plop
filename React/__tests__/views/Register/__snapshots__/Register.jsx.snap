// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="panel-header panel-header-sm"
  />
  <div
    className="content"
  >
    <div
      className="justify-content-center row"
    >
      <div
        className="justify-content-center col-12 col-md-4"
      >
        <div
          className="card-chart card"
          style={
            Object {
              "height": "300px",
            }
          }
        >
          <div
            className="card-header"
            style={
              Object {
                "paddingTop": "7%",
              }
            }
          >
            <h4
              className="card-title"
            >
              Register
            </h4>
          </div>
          <div
            className="card-body"
          >
            <div
              lg={2}
              md={2}
              style={
                Object {
                  "paddingTop": "2%",
                }
              }
              xs={10}
            >
              <div
                className="form-group"
              >
                <div>
                  E-mail
                </div>
                <div
                  className="input-group"
                >
                  <input
                    className="form-control"
                    id="register_email"
                    name="Email"
                    required={true}
                    type="email"
                  />
                </div>
              </div>
              <div
                className="form-group"
              >
                <div>
                  Password
                </div>
                <div
                  className="input-group"
                >
                  <input
                    className="form-control"
                    id="register_password"
                    name="Password"
                    required={true}
                    type="password"
                  />
                </div>
              </div>
              <div
                className="form-group"
              >
                <div>
                  Confirm password
                </div>
                <div
                  className="input-group"
                >
                  <input
                    className="form-control"
                    id="register_confirm_password"
                    name="Password"
                    required={true}
                    type="password"
                  />
                </div>
              </div>
              <div
                className="form-group"
              >
                <div>
                  ID number
                </div>
                <div
                  className="input-group"
                >
                  <input
                    className="form-control"
                    id="register_id_number"
                    name="id"
                    required={true}
                    type="number"
                  />
                </div>
              </div>
              <div
                className="form-group"
              >
                <div>
                  Gender
                </div>
                <div
                  className="input-group"
                >
                  <input
                    className="form-control"
                    id="register_id_number"
                    name="id"
                    required={true}
                    type="number"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="justify-content-right text-right card-footer"
            style={
              Object {
                "marginTop": "-2%",
              }
            }
          >
            <button
              className="btn btn-primaryBlue"
              onClick={[Function]}
              style={
                Object {
                  "width": "100px",
                }
              }
              type={undefined}
            >
              register
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
